src/qibolab/instruments/bluefors.py

```diff
@@ -0,0 +1,86 @@
+import json
+import re
+import socket
+
+from qibo.config import log
+
+# from datetime import datetime
+
+
+class TemperatureController:
+    """Bluefors temperature controller."""
+
+    def __init__(self, ip_address: str, port: int = 8888):
+        """Creation of the controller object.
+
+        Args:
+            ip_address (str): IP address of the board sending cryo temperature data.
+            port (int): port of the board sending cryo temperature data.
+        """
+        self.ip_address = ip_address
+        self.port = port
+        self.client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.is_connected = False
+
+    def connect(self):
+        """Connect to the socket."""
+        if self.is_connected:
+            return
+        try:
+            self.client_socket.connect((self.ip_address, self.port))
+            self.is_connected = True
+            log.info("Bluefors Temperature Controller Connected")
+        except ConnectionRefusedError as e:
+            log.error(f"Refused connection. IP: {self.ip_address} Port: {self.port}")
+            raise e
+        except TimeoutError as e:
+            log.error(f"Connection timed out. IP: {self.ip_address} Port: {self.port}")
+            raise e
+        except OSError as e:
+            log.error(
+                f"Cannot connect to host. IP: {self.ip_address} Port: {self.port}"
+            )
+            raise e
```

Comment:
Why not to keep the original errors as they are?

Reply:
It was just to avoid things that are not strictly needed. Since the method is otherwise simple, we could keep it simple.

However, if you want to keep the log, you could even catch multiple exceptions together.
```suggestion
        except (ConnectionRefusedError, TimeoutError, OSError) as e:
            log.error(f"Bluefors connection. IP: {self.ip_address} Port: {self.port}")
            raise e
```

But you could even log before attempting the connection.